src/ape_ethereum/provider.py

```diff
@@ -1116,7 +1116,17 @@ def _handle_execution_reverted(
                 **params,
             )
         )
-        return self.compiler_manager.enrich_error(result)
+        enriched = self.compiler_manager.enrich_error(result)
+
+        # Show call trace if availble
+        if enriched.txn:
+            # Apparently this is not always a receipt on a failed tx
+            if isinstance(enriched.txn, TransactionAPI) and enriched.txn.receipt:
+                enriched.txn.receipt.show_trace()
```

Comment:
when using ape-vyper, it already shows the source-traceback (the vyper lines last ran before the revert). How does this look when used with that feature? (im about to test and see for myself)

Reply:
answered my own question, I think it looks pretty good when together:

```
SUCCESS: Contract 'VyperContract' deployed to: 0x5FbDB2315678afecb367f032d93F642f64180aa3
Call trace for '0xa438858260fca5d95d2e2c0a6e37f9dea771c9738b102d8ebad696fa0c997490'
tx.origin=0x70997970C51812dc3A010C7d01b50e0d17dc79C8
VyperContract.setNumber(num=123) [23728 gas]

  File "/Users/jules/ApeProjects/ape-playground/scripts/rev.py", line 8, in main
    contract.setNumber(123, sender=notme)
  File "/Users/jules/ApeProjects/ape-playground/contracts/VyperContract.vy", line 84, in 
setNumber
    assert msg.sender == self.owner, "!authorized"
    ^^^^^^^^^^^^^^^^^^^^^^^

ERROR: (ContractLogicError) !authorized
```

You can see first your call trace gets printed, than the normal source traceback, and then the error (containing the revert message)